app/models/odk/dynamic_pattern_parser.rb

```diff
@@ -3,8 +3,8 @@
 module ODK
   # Abstract parent class for classes that parse $-style patterns for ODK.
   class DynamicPatternParser
-    # Basic regex for codes like $Question7
-    CODE_REGEX = /[$]!?[A-z]\w+/.freeze
+    # Basic regex for codes like $Question7 or $Question3:value
+    CODE_REGEX = /[$]!?[A-z]\w+:value|[$]!?[A-z]\w+/.freeze
```

Comment:
Tip: You can simplify to `/[$]!?[A-z]\w*(:value)?/`

Nice tool that validates and explains each and every symbol in your expression: https://regex101.com/

Reply:
Separate issue I just noticed: this doesn't actually support single-character question codes like `$x`, so we should also replace the `+` with `*` (this is already done above in my modified suggestion).

Reply:
Yup, i was using that tool!

Cool, i couldn't find a simpler OR operator, so yay for simplification. thank you!

Reply:
i tried your suggestion, but it didn't work. i'm also using this tool to ensure i'm using the ruby regex https://rubular.com/

Reply:
ah learned I needed a non-capturing group!